### PR TITLE
OMERO 5.6.0 / Python 3 updates

### PR DESCRIPTION
**Pull request by joshmoore**
Migrate changes from py3-ci back to the main repo. These will be merged into latest-ci for testing.

### CODE REVIEW

File: home/jobs/OMERO-build/config.xml [hunk 1]
```diff
@@ -63,7 +63,8 @@ export OMERO_BRANCH=$MERGE_PUSH_BRANCH
 
 virtualenv $WORKSPACE/omero-virtualenv --system-site-packages
 source $WORKSPACE/omero-virtualenv/bin/activate
-pip install -U setuptools
+pip install https://github.com/ome/zeroc-ice-py-manylinux/releases/download/0.1.0/zeroc_ice-3.6.5-cp36-cp36m-manylinux2010_x86_64.whl
```
Comment (mtbc) [inline home/jobs/OMERO-build/config.xml]:
```suggestion
pip install https://github.com/ome/zeroc-ice-py-manylinux/releases/download/0.2.0/zeroc_ice-3.6.5-cp36-cp36m-manylinux2010_x86_64.whl
```
and elsewhere, no idea if it's any better though!

File: home/jobs/OMERO-server/config.xml [hunk 2]
```diff
@@ -132,16 +154,19 @@ if [ &quot;$PURGE_DATA&quot; = &quot;true&quot; ]; then
     omego db init --serverdir $OMERO_DIST --dbname $OMERO_DB_NAME
 else
     echo &quot;Skipping PURGE_DATA:CleanDbAndRepo. Check for DB upgrade&quot;
-    omego db upgrade --serverdir $OMERO_DIST --dbname $OMERO_DB_NAME
+    omego db -vvv upgrade --serverdir $OMERO_DIST --dbname $OMERO_DB_NAME
```
Comment (mtbc) [inline home/jobs/OMERO-server/config.xml]: Want to keep it for now or leftover debugging?

File: home/jobs/OMERO-training/config.xml [hunk 3]
```diff
@@ -30,42 +50,36 @@ rm -rf $WORKSPACE/*</command>
...
 mv $WORKSPACE/$DIST $OMERO_EXAMPLE_DIST
 
 # UNZIP SERVER
-OMERO_SERVER_DIST=$WORKSPACE/OMERO.server
+export OMERODIR=$WORKSPACE/OMERO.server
 
 ZIP=$(ls OMERO.server*.zip)
 DIST=${ZIP%.zip}
-unzip $ZIP
+unzip -n $ZIP
```
Comment (mtbc) [inline home/jobs/OMERO-training/config.xml]: Interesting, does it help enable local overrides maybe?

**Comment by mtbc** [comment]
Looking generally good to me, I don't see any blockers.

File: home/jobs/OMERO-build/config.xml (hunk 1 above)
Comment (manics) [reply home/jobs/OMERO-build/config.xml]: The `manylinux2010` wheel bundle a static copy of openssl, so `0.2.0` will have been recompiled against a newer version.

Comment (joshmoore) [reply home/jobs/OMERO-build/config.xml]: @manics: i.e. better to bump or better to omit?

Comment (manics) [reply home/jobs/OMERO-build/config.xml]: If you use https://github.com/ome/zeroc-ice-py-centos7/ the wheel will be dynamically linked to the current openssl library which is safer.

File: home/jobs/OMERO-test-integration/config.xml [hunk 4]
```diff
@@ -13,23 +13,34 @@
   <triggers/>
   <concurrentBuild>false</concurrentBuild>
   <builders>
+    <hudson.tasks.Shell>
+      <command>if [ ! -e $WORKSPACE/.venv3 ]; then
+    virtualenv $WORKSPACE/.venv3 --system-site-packages
```
Comment (manics) [inline home/jobs/OMERO-test-integration/config.xml]:
```suggestion
    virtualenv $WORKSPACE/.venv3
```

File: home/jobs/OMERO-test-integration/config.xml [hunk 5]
```diff
@@ -13,23 +13,34 @@
...
+fi
+
+source $WORKSPACE/.venv3/bin/activate
+pip install https://github.com/ome/zeroc-ice-py-manylinux/releases/download/0.1.0/zeroc_ice-3.6.5-cp36-cp36m-manylinux2010_x86_64.whl
```
Comment (manics) [inline home/jobs/OMERO-test-integration/config.xml]:
```suggestion
pip install https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.2.1/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl
```

File: home/jobs/OMERO-web/config.xml [hunk 6]
```diff
@@ -76,103 +82,118 @@ then
...
-$OMERO_DIST/bin/omero config append omero.web.ui.top_links &apos;[&quot;Figure&quot;, &quot;figure_index&quot;, {&quot;title&quot;: &quot;Open Figure in new tab&quot;, &quot;target&quot;: &quot;_blank&quot;}]&apos;
-$OMERO_DIST/bin/omero config append omero.web.open_with &apos;[&quot;omero_figure&quot;, &quot;new_figure&quot;, {&quot;supported_objects&quot;:[&quot;images&quot;], &quot;target&quot;: &quot;_blank&quot;, &quot;label&quot;: &quot;OMERO.figure&quot;}]&apos;
+omero config append omero.web.apps &apos;&quot;omero_figure&quot;&apos;
+omero config append omero.web.ui.top_links &apos;[&quot;Figure&quot;, &quot;figure_index&quot;, {&quot;title&quot;: &quot;Open Figure in new tab&quot;, &quot;target&quot;: &quot;_blank&quot;}]&apos;
+omero config append omero.web.open_with &apos;[&quot;omero_figure&quot;, &quot;new_figure&quot;, {&quot;supported_objects&quot;:[&quot;images&quot;], &quot;target&quot;: &quot;_blank&quot;, &quot;label&quot;: &quot;OMERO.figure&quot;}]&apos;
+
+
+# OMERO.parade installation
+pip install https://github.com/snoopycrimecop/omero-parade/archive/$MERGE_PUSH_BRANCH.tar.gz
```
Comment (jburel) [inline home/jobs/OMERO-web/config.xml]: is there a reason not to follow the same installation step for parade than for the other app
e.g.
``pip install git+git://github.com/snoopycrimecop/omero-fpbioimage.git@$MERGE_PUSH_BRANCH#egg=omero-fpbioimage``

File: home/jobs/OMERO-web/config.xml [hunk 7]
```diff
@@ -76,103 +82,118 @@ then
...
 # OMERO.weberror installation
 pip install git+git://github.com/snoopycrimecop/omero-weberror.git@$MERGE_PUSH_BRANCH#egg=omero-weberror
-$OMERO_DIST/bin/omero config append omero.web.apps &apos;&quot;omero_weberror&quot;&apos;
+omero config append omero.web.apps &apos;&quot;omero_weberror&quot;&apos;
+
+# OMERO.webtest
+pip install git+git://github.com/snoopycrimecop/omero-webtest.git@$MERGE_PUSH_BRANCH#egg=omero-webtest
+omero config append omero.web.apps &apos;&quot;omero_webtest&quot;&apos;
 
 # OMERO.omero-webtagging-autotag installation
```
Comment (jburel) [inline home/jobs/OMERO-web/config.xml]: note indicating that solely the release version is installed for webtagging

File: home/jobs/OMERO-web/config.xml (hunk 6 above)
Comment (joshmoore) [reply home/jobs/OMERO-web/config.xml]: I don't know. There have been several editors of those sections.

File: home/jobs/OMERO-build/config.xml [hunk 8]
```diff
@@ -63,7 +63,8 @@ export OMERO_BRANCH=$MERGE_PUSH_BRANCH
 
 virtualenv $WORKSPACE/omero-virtualenv --system-site-packages
 source $WORKSPACE/omero-virtualenv/bin/activate
-pip install -U setuptools
+pip install https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.2.1/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl
+pip install -U setuptools &apos;omero-py&gt;=5.6.0.dev9&apos;
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply home/jobs/OMERO-build/config.xml]:
```suggestion
pip install -U setuptools &apos;omero-py&gt;=5.6.0&apos;
```

File: home/jobs/OMERO-push/config.xml [hunk 9]
```diff
@@ -109,4 +109,4 @@ git push -f git@github.com:$user/openmicroscopy.git HEAD:refs/heads/$MERGE_PUSH_
   <buildWrappers>
     <hudson.plugins.timestamper.TimestamperBuildWrapper plugin="timestamper@1.9"/>
   </buildWrappers>
-</project>
+</project>
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply home/jobs/OMERO-push/config.xml]: Not sure this is worth adding manually. These files are under Jenkins' control.

File: home/jobs/OMERO-server/config.xml [hunk 10]
```diff
@@ -41,22 +41,34 @@
...
+    virtualenv $WORKSPACE/.venv3
+fi
+
+source $WORKSPACE/.venv3/bin/activate
+pip install https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.2.1/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl
+pip install -U pip future
+pip install -U omego
+pip install tables
+pip install jinja2
+pip install reportlab
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply home/jobs/OMERO-server/config.xml]: Would it be cleaner to install figure and have them pulled in?

[the remark this answers is not in the document]
Comment (jburel) [reply home/jobs/OMERO-server/config.xml]: Probably but I don't think it is necessary to add more work at this point.
a simple comment will be enough

File: home/jobs/OMERO-web/config.xml (hunk 6 above)
Comment (jburel) [reply home/jobs/OMERO-web/config.xml]: @will-moore any reason?

Comment (will-moore) [reply home/jobs/OMERO-web/config.xml]: No reason that I can remember

**Comment by joshmoore** [comment]
I'm unsure of the state of the devspaces at this point. Since changes here would need to be rolled out to all the CIs to prevent diffs, I'm going to merge as is. I assume there are other cleanups that would be appropriate now that the bulk of the Python 3 work is done.

**Comment by sbesson** [comment]
Worth merging #160 and tagging all of this as `0.15.0` ?

**Comment by joshmoore** [comment]
That would work for me, though my caveat of not knowing the current state of all 4 CIs still stands. That being said, it could easily be 0.15.1 or whatever.

**Pull request merged**